Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,294 @@
+        }
+
+        private static decimal FindHighestBalance(string[] peopleAndBalances, StringBuilder PersonWithHighestBalance)
+        {
+            var HighestBalanceEver = decimal.MinValue;
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                try
+                {
+                    var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+                    
+                    
+                    if (highestAmountOfPerson == HighestBalanceEver)
+                    {
+                        PersonWithHighestBalance.Append(", ");
+                        PersonWithHighestBalance.Append(currentPersonData[0]);
+                    }
+
+                    if (highestAmountOfPerson > HighestBalanceEver)
+                    {
+                        HighestBalanceEver = highestAmountOfPerson;
+                        var currentOne = PersonWithHighestBalance.ToString();
+
+                        if (string.IsNullOrEmpty(currentOne))
+                        {
+                            PersonWithHighestBalance.Append(currentPersonData[0]);
+                        }
+                        else
+                        {
+                            PersonWithHighestBalance.Clear(); 
+                            PersonWithHighestBalance.Append(currentPersonData[0]);
+                        }
+                    }
+                }
+                catch (ArgumentNullException)
+                {
+                    Console.WriteLine($"There was a person with no balance in the file, namely : {currentPersonData[0]}");
+                    break;
+                }
+            }
+
+            return HighestBalanceEver;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestLossEver = decimal.MinValue;
+            var PersonWithHighestLoss = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var allAmounts = currentPersonData[1..];
+
+                // calculateLoss
+
+                if (allAmounts.Length <= 1)
+                {
+                    return InValidOutput;
+                }
+
+                for (int j = 0; j < allAmounts.Length - 1; j++)
+                {
+                    try
+                    {
+                        var amount1 = decimal.Parse(allAmounts[j], NumberStyles.Currency | NumberStyles.Number);
+                        try
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4967975)

Reply:
Sorry, I do not see how I can make this work and  not making the code looks less readable